network/dag/state.go

```diff
@@ -43,17 +44,14 @@ const (
 
 // State has references to the DAG and the payload store.
 type state struct {
-	db                               stoabs.KVStore
-	graph                            *dag
-	payloadStore                     PayloadStore
-	transactionalObservers           []Observer
-	nonTransactionalObservers        []Observer
-	transactionalPayloadObservers    []PayloadObserver
-	nonTransactionalPayloadObservers []PayloadObserver
-	txVerifiers                      []Verifier
-	notifiers                        map[string]Notifier
-	xorTree                          *treeStore
-	ibltTree                         *treeStore
+	db               stoabs.KVStore
+	graph            *dag
+	payloadStore     PayloadStore
+	txVerifiers      []Verifier
+	notifiers        map[string]Notifier
+	xorTree          *treeStore
+	ibltTree         *treeStore
+	transactionCount prometheus.Counter
```

Comment:
we now have 3 types of diagnostical data, diagnostics, `State`'s `func (s *state) Statistics(ctx context.Context) Statistics` and internal prometheus counters.

Maybe Statistics should be merged with Diagnostics?

Reply:
Statistics removed